File: Subprogramas/Polimorfismo/Polimorfismo.java
//Java

class Polimorfismo {
    void imprimir(String s) {
        System.out.println(s);
    }

    void imprimir(int i) {
        System.out.println(i);
    }
}

public class Main {
    public static void main(String[] args) {
        Polimorfismo p = new Polimorfismo();
        p.imprimir("Hello");  // chama o método que aceita String
        p.imprimir(10);       // chama o método que aceita int
    }
}

File: Subprogramas/Polimorfismo/Polimorfismo.pas
//Pascal

program Polimorfismo;

procedure Imprimir(X: Integer);
begin
    writeln(X);
end;

procedure Imprimir(X: String);
begin
    writeln(X);
end;

begin
    Imprimir(10);       { chama o Imprimir que aceita Integer }
    Imprimir('Hello');  { chama o Imprimir que aceita String }
end.
